src/components/Button.tsx

```diff
@@ -23,36 +25,46 @@ export function Button({
   size = 'S',
   text,
   type = 'primary',
+  inheritColor = false,
   submit = false,
+  title,
   onClick,
   ...rest
 }: ButtonProps) {
   return (
     <button
       aria-disabled={disabled}
       className={cx(
-        'flex items-center gap-2 rounded-default px-2 py-2',
+        'flex items-center rounded-default px-2 py-2',
         {
           'w-full': fluid,
         },
         {
-          'justify-center px-5 py-3': size === 'M',
+          'gap-2': !size,
+          'justify-center gap-2 px-5 py-3': size === 'M',
+          'justify-center gap-1 px-[.5rem] py-[.25rem]': size === 'S',
         },
         {
           'pl-3': !!Action && actionPosition === 'left',
           'pr-3': !!Action && actionPosition === 'right',
+          'pl-[.25rem]': !!Action && actionPosition === 'left' && size === 'S',
+          'pr-[.25rem]': !!Action && actionPosition === 'right' && size === 'S',
         },
         {
-          'bg-btn-bg-primary-default text-btn-txt-primary-default': type === 'primary',
-          'bg-btn-bg-primary-disabled text-btn-txt-primary-disabled': disabled && type === 'primary',
+          'bg-btn-bg-primary-default text-btn-txt-primary-default': !inheritColor && type === 'primary',
+          'bg-btn-bg-primary-disabled text-btn-txt-primary-disabled': !inheritColor && disabled && type === 'primary',
+          'bg-btn-bg-primary-default': inheritColor && type === 'primary',
         },
         {
-          'bg-btn-bg-secondary-default text-btn-txt-secondary-default': type === 'secondary',
-          'bg-btn-bg-secondary-disabled text-btn-txt-secondary-disabled': disabled && type === 'secondary',
+          'bg-btn-bg-secondary-default text-btn-txt-secondary-default': !inheritColor && type === 'secondary',
+          'bg-btn-bg-secondary-disabled text-btn-txt-secondary-disabled':
+            !inheritColor && disabled && type === 'secondary',
+          'bg-btn-bg-secondary-default': inheritColor && type === 'secondary',
```

Comment:
We need a more robust design system that contains button styles for all the buttons we need. I think this is ok for now but I agree it could be better. inheritColor allows the text color to be pass as className prop.